flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/utils/LookupJoinUtil.java

```diff
@@ -197,92 +363,177 @@ public static int[] getOrderedLookupKeys(Collection<Integer> allLookupKeys) {
         return lookupKeyIndicesInOrder.stream().mapToInt(Integer::intValue).toArray();
     }
 
+    public static AsyncLookupOptions getMergedAsyncOptions(
```

Comment:
It's better we can add some UTs the verify the logica

Reply:
convered by AsyncLookupOptionsTest#testAsyncLookupOptions